### PR TITLE
[jemalloc] Support non-Windows

### LINKED ISSUE #17409: [jemalloc] build failure, "versions" feature can't recognize port's current version format 4.3.1-4 in "overrides" section
**Host Environment**
- OS: Mac
- Compiler: apple clang 12.0

**To Reproduce**
`./vcpkg install --feature-flags="manifests" --feature-flags="versions"`

**Failure logs** 
`
    $.overrides[3] (an override): 'version' text was not a relaxed version:
Error: String `4.3.1-4` must only contain dot-separated numeric values without leading zeroes.
See https://github.com/Microsoft/vcpkg/tree/master/docs/users/manifests.md for more information.
`

**Additional context**
with the folling value in overrides section of any vcpkg.json file
`
"overrides": [
{
        "name": "jemalloc", "version": "4.3.1-4"
}
]

### PR DESCRIPTION
**Pull request by JackBoosY**
Now the upstream cmake code doesn't support non-Windows, add support now.

Upstream PR: https://github.com/jemalloc/jemalloc-cmake/pull/4

Fixes #17409

### CODE REVIEW

File: ports/jemalloc/add-unix-support.patch [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
++"#ifdef _WIN32\n"
+ "int result;\n"
+ "#ifdef _WIN32\n"
+ "SYSTEM_INFO si;\n"
+@@ -740,6 +732,9 @@ file(WRITE ${SRC}
+ "result = sysconf(_SC_PAGESIZE);\n"
+ "#endif\n"
+ "printf(\"%d\", result);\n"
++"#else\n"
++"printf(\"page size = %d\\n\", getpagesize());\n"
```
Comment (JackBoosY) [inline ports/jemalloc/add-unix-support.patch:79]:
```suggestion
+"printf(\"%d\", getpagesize());\n"
```

File: ports/jemalloc/add-unix-support.patch [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
++"#endif\n"
+ "return 0;\n"
+ "}\n"
```
Comment (JackBoosY) [inline ports/jemalloc/add-unix-support.patch:82]:
```
"#include <stdio.h>\n"
"#ifdef _WIN32\n"
"#include <windows.h>\n"
"#elif defined(__APPLE__)\n"
"#include <sys/param.h>\n"
"#include <sys/sysctl.h>\n"
"#else\n"
"#include <unistd.h>\n"
"#endif\n"
"int main(int argc, const char** argv) {\n"
"#ifdef _WIN32\n"
"int result;\n"
"SYSTEM_INFO si;\n"
"GetSystemInfo(&si);\n"
"result = si.dwPageSize;\n"
"result = sysconf(_SC_PAGESIZE);\n"
"printf(\"%d\", result);\n"
"#elif defined(__APPLE__)\n"
"int mib[2], value, pagesize;\n"
"size_t size;\n"
"mib[0] = CTL_HW;\n"
"mib[1] = HW_PAGESIZE;\n"
"size = sizeof value;\n"
"if (sysctl(mib, 2, &value, &size, NULL, 0) == -1)\n"
"    pagesize = -1;\n"
"pagesize = value;\n"
"printf(\"%d\\n\", pagesize);\n"
"#else\n"
"printf(\"%d\\n\", getpagesize());\n"
"#endif\n"
"return 0;\n"
"}\n"
```

**Comment by Neumann-A** [comment]
will create a merge conflict with #25009

**Comment by JackBoosY** [comment]
In favor of https://github.com/microsoft/vcpkg/pull/25009.

**Pull request closed without merging**